test/robot/web/grid.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/google/gapid/test/robot/job"
+	"github.com/google/gapid/test/robot/replay"
+	"github.com/google/gapid/test/robot/report"
+	q "github.com/google/gapid/test/robot/search/query"
+	"github.com/google/gapid/test/robot/trace"
+)
+
+type cell struct {
+	Scheduled int
```

Comment:
Don't use public fields with private structs unless you're doing horrible reflection stuff.

Reply:
This gets serialized to JSON, which does the horrible reflection stuff.